processor/metricstransformprocessor/metrics_transform_processor.go

```diff
@@ -92,9 +93,13 @@ func (f internalFilterStrict) getSubexpNames() []string {
 	return nil
 }
 
-func (f internalFilterStrict) labelMatched(metric *metricspb.Metric) bool {
+func (f internalFilterStrict) labelMatched(metric *metricspb.Metric) *metricspb.Metric {
+	metricWithMatchedLabel := proto.Clone(metric).(*metricspb.Metric)
```

Comment:
Why do you need to clone? Can you avoid it?

Reply:
Rather than cloning the whole metric object, now I am copying the Resource and MetricDescriptor. We don't want to modify the incoming pdata.Metric, thats way generating a new Metric and returning it. Pushed an update.

Reply:
@hossain-rayhan Was there any specific reasons to clone the objects. Like were you seeing an error or test failures ?
I am asking because cloning seems to be reason for why updates are not working with experimental_match_label. 
cc @anuraaga 

My observation is as follows - When we returns a cloned "MetricDescriptor" object, the update at this point fails - https://github.com/snowflakedb/opentelemetry-collector-contrib/blob/0250e4459e2b484a5cd6f2a740f8b7525124a401/processor/metricstransformprocessor/metrics_transform_processor.go#L424

This is because, the above line is updating a cloned object rather than updating the original "MetricDescriptor" object. 
one such example is here - https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7071

Please let me know what do you think ?